### DISCUSSION TITLE
树形选择器子级有相同数据时，会展示多条重复数据

### DISCUSSION THREAD

**Discussion by ycyyf** [ant-design/ant-design#46530] [Q&A]
使用树形选择器ProFormTreeSelect显示角色和用户两个层级，当一个用户出现在多个角色中，搜索张三然后选中后会出现这种情况：会重复显示多个
☑️超级管理员
  ☑️ 张三
  ☑️ 张三
  ☑️ 张三
  ☑️ 张三
  ☑️ 李四
☑️ 管理员
  ☑️ 张三

**Comment by Wxh16144** [comment]
上代码！

**Reply by ycyyf** [reply]
import { ProFormTreeSelect } from '@ant-design/pro-components';

export default () => {
  const treeData = [
    {
      code: 'a001',
      name: '超管',
      list: [
        {
          code: '001',
          name: '张三',
        },
        {
          code: '002',
          name: '李四',
        }
      ]
    },
    {
      code: 'a002',
      name: '超管2',
      list: [
        {
          code: '001',
          name: '张三',
        }
      ]
    },
    {
      code: 'a003',
      name: '超管3',
      list: [
        {
          code: '001',
          name: '张三',
        }
      ]
    },
    {
      code: 'a004',
      name: '超管4',
      list: [
        {
          code: '001',
          name: '张三',
        }
      ]
    }
  ]
  return (
    <ProFormTreeSelect
      name="handler"
      placeholder="请选择"
      allowClear
      width={330}
      secondary
      rules={[{ required: true, message: '请选择处理人' }]}
      fieldProps={{
        suffixIcon: null,
        filterTreeNode: true,
        showSearch: true,
        popupMatchSelectWidth: false,
        labelInValue: true,
        autoClearSearchValue: true,
        multiple: true,
        treeCheckable: true,
        treeNodeFilterProp: 'label',
        fieldNames: {
          label: 'name',
          value: 'code',
          children: 'list',
        },
        treeData
      }}
      label="处理人"
    />
  );
};

**Reply by yoyo837** [reply]
提供在线编辑的，这个还需要精力去整理运行，人人都这样贴，没精力应对。

**Comment by susanforme** [comment]
value要是唯一的, 实在不行,labelinvalue:true 最后取label就行

**Reply by ycyyf** [reply]
现在的场景value不是唯一的，会出现这种

**Comment by ycyyf** [comment]
重新为每个元素设置了唯一的key，目前可以了
